File: src/parser/v2/helpers/tree-add.js
"use strict";

/**
 * Adds NODE object to node tree.
 *
 * @param  {object} STATE - Main loop state object.
 * @param  {object} NODE - Object containing parsed information (node).
 * @return {undefined} - Nothing is returned.
 */
module.exports = (STATE, NODE) => STATE.tables.tree.nodes.push(NODE);

File: src/parser/nim/parser/helpers/tree_add.nim
import std/tables

import types

# Add node object to tree.
#
# @param  {object} S - State object.
# @param  {object} N - Node object.
# @return - Nothing is returned.
proc add*(S: State, N: Node) =
    S.tables.tree["nodes"].add(N)
